cli/specifier_handler.rs

```diff
@@ -187,23 +233,54 @@ impl FetchHandler {
 
     Ok(FetchHandler {
       disk_cache,
+      dynamic_permissions,
       file_fetcher,
-      permissions,
     })
   }
 }
 
 impl SpecifierHandler for FetchHandler {
-  fn fetch(&mut self, requested_specifier: ModuleSpecifier) -> FetchFuture {
-    let permissions = self.permissions.clone();
+  fn fetch(
+    &mut self,
+    requested_specifier: ModuleSpecifier,
+    maybe_location: Option<Location>,
+    is_dynamic: bool,
+  ) -> FetchFuture {
+    // When the module graph fetches dynamic modules, the set of dynamic
+    // permissions need to be applied.  Other static imports have all
+    // permissions.
+    let permissions = if is_dynamic {
+      self.dynamic_permissions.clone()
+    } else {
+      Permissions::allow_all()
+    };
```

Comment:
This is indeed cleaner. We could bike shed on the naming, but I'm not gonna do that right now - although it might be a misleading in case of creates module graph for worker - it is another top level module (not dynamic), but it should use runtime permissions - and I think that's already happening.

Reply:
Yeah, but not in the module graph...  There is still a bit of work to do to try consolidate where we have logic like this.